fnl/nfnl/core.fnl

```diff
@@ -243,9 +243,12 @@
         (f:close)
         content)))
 
-(fn spit [path content]
-  "Spit the string into the file."
-  (match (io.open path "w")
+(fn spit [path content opts]
+  "Spit the string into the file. When opts.append is true, add to the file."
+  (var mode "w")
+  (when (and opts opts.append)
+    (set mode "a"))
+  (match (io.open path mode)
```

Comment:
I love the idea however I'd like this to be written as:

```fennel
(io.open path
  (if (get opts :append)
    "a"
    "w"))
```

I personally think that's easier to follow but happy to listen to feedback! I would do this change myself, but I want to make sure you get the attribution 😄 

(I've been on holiday / working / sprained ankle / maybe coming down with something so sorry for the delay!)

Reply:
And I'm using `get` here because it's safe to use with `nil` etc.

Reply:
Yes, that's a much better way of expressing the setting of the mode of the opened stream. `get` is defined before `spit` so `spit` can leverage its safety.

Thanks!